Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm

```diff
@@ -179,88 +180,58 @@
             return lighterGray;
         });
     case WarningItem::TitleText:
-    case WarningItem::ExclamationPoint:
+    case WarningItem::WarningSymbol:
         return WTF::switchOn(warningView.warning->data(), [&] (const WebKit::BrowsingWarning::SafeBrowsingWarningData&) {
             return narrow ? white : red;
         }, [&] (const WebKit::BrowsingWarning::HTTPSNavigationFailureData&) {
-            return !narrow && item == WarningItem::TitleText ? white : red;
+            return item == WarningItem::TitleText ? darkGray : red;
         });
     case WarningItem::MessageText:
     case WarningItem::ShowDetailsButton:
     case WarningItem::ContinueButton:
         return WTF::switchOn(warningView.warning->data(), [&] (const WebKit::BrowsingWarning::SafeBrowsingWarningData&) {
             return narrow ? white : [UIColor darkTextColor];
         }, [&] (const WebKit::BrowsingWarning::HTTPSNavigationFailureData&) {
-            return item == WarningItem::MessageText ? mediumGray : white;
+            return darkGray;
         });
     case WarningItem::GoBackButton:
         return WTF::switchOn(warningView.warning->data(), [&] (const WebKit::BrowsingWarning::SafeBrowsingWarningData&) {
             return narrow ? white : warning.tintColor;
         }, [&] (const WebKit::BrowsingWarning::HTTPSNavigationFailureData&) {
-            return white;
+            return narrow ? darkGray : warning.tintColor;
         });
     }
 #endif
     ASSERT_NOT_REACHED();
     return nil;
 }
 
-@interface WKSafeBrowsingExclamationPoint : ViewType
-@end
-
-@implementation WKSafeBrowsingExclamationPoint
-
-- (void)drawRect:(RectType)rect
+static ViewType* viewForIconImage(_WKWarningView *warningView)
 {
-    constexpr CGFloat centerX = exclamationPointSize / 2;
-    constexpr CGFloat pointCenterY = exclamationPointSize * 7 / 30;
-    constexpr CGFloat pointRadius = 2.25 * exclamationPointSize / 30;
-    constexpr CGFloat lineBottomCenterY = exclamationPointSize * 13 / 30;
-    constexpr CGFloat lineTopCenterY = exclamationPointSize * 23 / 30;
-    constexpr CGFloat lineRadius = 1.75 * exclamationPointSize / 30;
-    ViewType *warning = self.superview.superview;
+    NSString *symbolName;
+    WebCore::CocoaColor *color = colorForItem(WarningItem::WarningSymbol, warningView);
+    Boolean shouldSetTint = NO;
```

Comment:
```suggestion
    BOOL shouldSetTint = NO;
```

Reply:
Whoops, I spent too much time in CF code before writing this.